### PR TITLE
Fix checking of comment node by using the hiddenTokenUtil

### PR DESCRIPTION
**Pull request by markdomeng**

### CODE REVIEW

File: com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/formatting/ExtendedFormattingConfigBasedStream.java [hunk 1]
```diff
@@ -727,7 +728,13 @@ public void setFormattingActive(final boolean formattingActive) {
   private boolean isCommentNode(final INode node) {
     if (node instanceof ILeafNode && node.getGrammarElement() != null) {
       EObject grammarElement = node.getGrammarElement();
-      return grammarElement.equals(formatter.getSLCommentRule()) || grammarElement.equals(formatter.getMLCommentRule());
+      if (grammarElement instanceof TerminalRule) {
```
Comment (rubenporras) [inline com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/formatting/ExtendedFormattingConfigBasedStream.java:731]: must always be a TerminalRule? Should we not keep the old code in an else branch?

Comment (andrewL-avlq) [reply com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/formatting/ExtendedFormattingConfigBasedStream.java:731]: The old code always returns false for non-TerminalRule elements as the comment rules are TerminalRules.

File: com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/formatting/ExtendedFormattingConfigBasedStream.java [hunk 2]
```diff
@@ -727,7 +728,13 @@ public void setFormattingActive(final boolean formattingActive) {
   private boolean isCommentNode(final INode node) {
     if (node instanceof ILeafNode && node.getGrammarElement() != null) {
       EObject grammarElement = node.getGrammarElement();
-      return grammarElement.equals(formatter.getSLCommentRule()) || grammarElement.equals(formatter.getMLCommentRule());
+      if (grammarElement instanceof TerminalRule) {
+        TerminalRule rule = (TerminalRule) grammarElement;
+        TerminalRule slCommentRule = formatter.getSLCommentRule();
+        TerminalRule mlCommentRule = formatter.getMLCommentRule();
+        return (rule.getName().equals(slCommentRule.getName()) && rule.isFragment() == slCommentRule.isFragment())
```
Comment (rubenporras) [inline com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/formatting/ExtendedFormattingConfigBasedStream.java:735]: what about writing the comparator once? I think the following code is nicer, what do you think?

      if (grammarElement instanceof TerminalRule) {
        TerminalRule rule = (TerminalRule) grammarElement;
        Comparator<TerminalRule> comparator = Comparator.comparing(TerminalRule::getName)//
            .thenComparing(TerminalRule::isFragment);
        return comparator.compare(formatter.getSLCommentRule(), rule) == 0 //
            || comparator.compare(formatter.getMLCommentRule(), rule) == 0;
      }

Comment (markdomeng) [reply com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/formatting/ExtendedFormattingConfigBasedStream.java:735]: ah this is nice. Im wondering tho, how deep should the comparison be?

Comment (rubenporras) [reply com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/formatting/ExtendedFormattingConfigBasedStream.java:735]: I am sorry, I do not know that either

File: com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/formatting/ExtendedFormattingConfigBasedStream.java [hunk 3]
```diff
@@ -727,7 +728,9 @@ public void setFormattingActive(final boolean formattingActive) {
   private boolean isCommentNode(final INode node) {
     if (node instanceof ILeafNode && node.getGrammarElement() != null) {
       EObject grammarElement = node.getGrammarElement();
-      return grammarElement.equals(formatter.getSLCommentRule()) || grammarElement.equals(formatter.getMLCommentRule());
+      if (grammarElement instanceof TerminalRule) {
+        return hiddenTokenHelper.isComment((TerminalRule) grammarElement);
```
Comment (andrewL-avlq) [inline com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/formatting/ExtendedFormattingConfigBasedStream.java:732]: I am ok with this approach. Anything to add Ruben?

Comment (rubenporras) [reply com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/formatting/ExtendedFormattingConfigBasedStream.java:732]: me too

**Review by rubenporras — APPROVED** [review]

**Pull request merged**